Don't rely on GNU regex extension for character classes.

$ echo --with-system-nspr --with-system-nss | gsed -e 's/--with-system-nspr\S* *//'
--with-system-nss
$ echo --with-system-nspr --with-system-nss | sed -e 's/--with-system-nspr\S* *//'
sed: 1: "s/--with-system-nspr\S* ...": RE error: trailing backslash (\)

diff --git old-configure.in old-configure.in
index 7722587fadac..acf5d8d67a60 100644
--- configure.in
+++ configure.in
@@ -4405,7 +4405,7 @@ ac_configure_args="$_SUBDIR_CONFIG_ARGS"
 
 # --with-system-nspr will have been converted into the relevant $NSPR_CFLAGS
 # and $NSPR_LIBS.
-ac_configure_args="`echo $ac_configure_args | sed -e 's/--with-system-nspr\S* *//'`"
+ac_configure_args="`echo $ac_configure_args | sed -e 's/--with-system-nspr[[^[:space:]]]* *//'`"
 
 if test "$_INTL_API" = no; then
     ac_configure_args="$ac_configure_args --without-intl-api"
